pkg/operator/oidcdiscoveryendpoint/controller.go

```diff
@@ -0,0 +1,424 @@
+package oidcdiscoveryendpoint
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	jose "gopkg.in/square/go-jose.v2"
+
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"github.com/openshift/cloud-credential-operator/pkg/aws"
+	"github.com/openshift/cloud-credential-operator/pkg/operator/platform"
+	awsannotator "github.com/openshift/cloud-credential-operator/pkg/operator/secretannotator/aws"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configset "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName                = "oidcdiscoveryendpoint"
+	deploymentName                = "cloud-credential-operator"
+	operatorNamespace             = "openshift-cloud-credential-operator"
+	tokenSignerConfigMapName      = "bound-sa-token-signing-certs"
+	tokenSignerConfigMapNamespace = "openshift-kube-apiserver"
+	credentialsSecretNamespace    = "openshift-cloud-credential-operator"
+	credentialsSecretName         = "cloud-credential-operator-s3-creds"
+	discoveryURI                  = ".well-known/openid-configuration"
+	keysURI                       = "keys.json"
+	discoveryTemplate             = `{
```

Comment:
we have bindata already loading assets, why not this template.

Reply:
my reasoning was that it was not a kube resources and it was small and easier to load and access/understand without layers of indirection.

Reply:
if you can add a note to the commit message on this, it will help in easy discoverability.  github comments aren't great after the fact.